utils/hwloc/misc.h

```diff
@@ -946,20 +947,69 @@ hwloc_utils_parse_cpuset_format(const char *string)
     return HWLOC_UTILS_CPUSET_FORMAT_HWLOC;
   else if (!strcmp(string, "list"))
     return HWLOC_UTILS_CPUSET_FORMAT_LIST;
+  else if (!strcmp(string, "systemd-dbus-api"))
+    return HWLOC_UTILS_CPUSET_FORMAT_SYSTEMD;
   else if (!strcmp(string, "taskset"))
     return HWLOC_UTILS_CPUSET_FORMAT_TASKSET;
   else
     return HWLOC_UTILS_CPUSET_FORMAT_UNKNOWN;
 }
 
+/* The AllowedCPUs systemd DBus API syntax expects a string as follows:
+ * "ay 0xNNNN 0xAA [0xBB [...]]"
+ * where:
+ *   "0xNNNN" is the size of the array, max size 2^26.
+ *   "0xAA [0xBB [...]]" is the cpuset mask given bytes after bytes, little endian order.
+ * e.g. the output of `hwloc-calc --cpuset-input-format list 0,31-32,63-64,77 --cpuset-output-format systemd-dbus-api` is
+ *   "AllowedCPUs ay 0x0a 0x01 0x00 0x00 x80 0x01 0x00 0x00 0x80 0x01 0x20
+ * */
+#define HWLOC_SYSTEMD_ALLOWEDCPUS_PREFIX_FORMAT "AllowedCPUs ay 0x%02x"
+#define HWLOC_SYSTEMD_ALLOWEDCPUS_PREFIX_SIZE 19
+#define HWLOC_SYSTEMD_ALLOWEDCPUS_BYTES_FORMAT " 0x%02x"
+#define HWLOC_SYSTEMD_ALLOWEDCPUS_BYTES_SIZE 5
+static __hwloc_inline int hwloc_utils_systemd_asprintf(char ** strp, const struct hwloc_bitmap_s * __hwloc_restrict set)
+{
+  int last = hwloc_bitmap_last(set);
+  if (last == -1 )
+    abort();
+  int bytes = last / 8 + 1; /* how many bytes to represent the cpuset bit mask */
+  int buflen = HWLOC_SYSTEMD_ALLOWEDCPUS_PREFIX_SIZE + HWLOC_SYSTEMD_ALLOWEDCPUS_BYTES_SIZE * bytes + 1;
+  int ret = 0;
+#ifdef HWLOC_SYSTEMD_ALLOWEDCPUS_DEBUG
```

Comment:
is there any way to activate this debug? should we keep it? or just make it a comment? some compiler will complain about fprintf is the middle of variable declarations

Reply:
Nobody's ever going to enable them, I'd remove them and just make sure we have tests for different cases in make check.

Reply:
done.